op-chain-ops/genesis/layer_two.go

```diff
@@ -46,3 +54,74 @@ func BuildL2DeveloperGenesis(config *DeployConfig, l1StartBlock *types.Block) (*
 
 	return db.Genesis(), nil
 }
+
+// BuildL2MainnetGenesis will build an L2 Genesis suitable for a Superchain mainnet that does not
+// require a pre-bedrock migration & supports optional governance token predeploy.
+func BuildL2MainnetGenesis(config *DeployConfig, l1StartBlock *types.Block) (*core.Genesis, error) {
+	genspec, err := NewL2Genesis(config, l1StartBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	db := state.NewMemoryStateDB(genspec)
+
+	storage, err := NewL2StorageConfig(config, l1StartBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	immutable, err := NewL2ImmutableConfig(config, l1StartBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set up the proxies
+	depBytecode, err := bindings.GetDeployedBytecode("Proxy")
+	if err != nil {
+		return nil, err
+	}
+	if len(depBytecode) == 0 {
+		return nil, errors.New("Proxy has empty bytecode")
+	}
+	for i := uint64(0); i <= 2048; i++ {
+		bigAddr := new(big.Int).Or(bigL2PredeployNamespace, new(big.Int).SetUint64(i))
+		addr := common.BigToAddress(bigAddr)
+		db.CreateAccount(addr)
+		db.SetCode(addr, depBytecode)
+		db.SetState(addr, AdminSlot, predeploys.ProxyAdminAddr.Hash())
+	}
+
+	// Set up the implementations
+	deployResults, err := immutables.BuildOptimism(immutable)
+	if err != nil {
+		return nil, err
+	}
+	for name, predeploy := range predeploys.Predeploys {
+		addr := *predeploy
+		if predeploys.IsDeprecated(addr) {
+			continue
+		}
+		if addr == predeploys.GovernanceTokenAddr && storage["GovernanceToken"] == nil {
```

Comment:
I don't love how this is implicit on an upstream function mutating state, perhaps we can use the bool on the deploy config itself instead here?

Reply:
I think it would be good to have a logline saying governance disabled or something

Reply:
yes should have used the config in the first place: fixed.  Also added a log line.